konrad/aerosol.py

```diff
@@ -0,0 +1,289 @@
+    aerosol_type: choose between 'no_aerosol' for no volcanic aerosols
+                                 'all_aerosol_properties' for volcanic aerosols,
+                                      described by LW (ext) and SW (ext, g, ssa)
+                                  'ecmwf' for volcanic aerosols, described by LW
+                                      (ext) and SW (ext at 550 mu m) (currently
+                                      not implemented in version available online)
+    aerosolLevelShiftInput: choose the shift of the aerosol layer relative to the
+                         original Pinatubo aerosol layer (units in km)
+    monthsAfterEruption: chose the forcing data to describe the aerosol characteristics
+                         x months after the eruption. Mount Pinatubo erupted on June 15, 1991.
+                         allowed inputs in [0,18]
+                         the input starts at 1 (July 1991).
+                         Default value 2 is August 1991.
+                         (not implemented yet)
+    includeSWForcing: if True: imports the SW forcing files and calculates the radiative
+                         fluxes using them
+    includeScattering: includes SW scattering component
+                        (only for 'all_aerosol_properties')
+    includeScattering: includes SW absorption component
+                        (only for 'all_aerosol_properties')
+    includeLWForcing: if True: imports the LW forcing files and calculates the radiative
+                         fluxes using them
+"""
+ 
+import os
+import abc
+import xarray as xr
+import scipy as sc
+import numpy as np
+import typhon.physics as ty
+from sympl import DataArray
+
+def get_aerosol_waveband_data_array(values, units='dimensionless', numlevels=200,
+                            sw=True):
+    """Return a DataArray of values."""
+    if isinstance(values, DataArray):
+        return values
+
+    if sw:
+        dims_bands = 'num_shortwave_bands'
+        numbands = 14
+    else:
+        dims_bands = 'num_longwave_bands'
+        numbands = 16
+
+    if isinstance(values, (int, float)):
+        return DataArray(values * np.ones((numlevels, numbands)),
+                         dims=('mid_levels', dims_bands),
+                         attrs={'units': units}).T
+
+    elif isinstance(values, np.ndarray):
+        if values.shape == (numlevels,):
+            return DataArray(
+                np.repeat(values[:, np.newaxis], numbands, axis=1),
+                dims=('mid_levels', dims_bands),
+                attrs={'units': units},
+            ).T
+        elif values.shape == (numlevels, numbands):
+            return DataArray(
+                values,
+                dims=('mid_levels', dims_bands),
+                attrs={'units': units},
+            ).T
+
+    raise TypeError(
+        'Aerosol variable input must be a single value, `numpy.ndarray` or a '
+        '`sympl.DataArray`')
+
+class Aerosol(metaclass=abc.ABCMeta):
```

Comment:
If you want the aerosols to be part of the output netCDF file, we can talk about making it a `konrad.Component` in the future :relaxed: